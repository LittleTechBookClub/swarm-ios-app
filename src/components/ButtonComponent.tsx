import React from 'react';
import { useFonts } from 'expo-font';
import { Text, StyleSheet, Pressable } from 'react-native';

type Props = {
  text: string;
  handleClick: () => void;
};

const ButtonComponent: React.FC<Props> = ({ text, handleClick }: Props) => {
  const [fontsLoaded] = useFonts({
    'Roboto-Medium': require('../../assets/fonts/Roboto/Roboto-Medium.ttf'),
  });

  if (!fontsLoaded) {
    return <Text>Loading...</Text>;
  }

  return (
    <Pressable
      style={({ pressed }) => [
        {
          backgroundColor: pressed ? '#FFFFFF' : '#000000',
          color: pressed ? '#000000' : '#FFFFFF',
        },
        styles.buttonContainer,
      ]}
      onPress={handleClick}
    >
      {text}
    </Pressable>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    width: '100%',
    height: 56,
    borderRadius: 8,
    paddingVertical: 16,
    paddingHorizontal: 40,
    borderColor: '#000000',
    borderWidth: 1,
    fontFamily: 'Roboto-Medium',
    fontSize: 16,
    textAlign: 'center',
  },
});
export default ButtonComponent;
