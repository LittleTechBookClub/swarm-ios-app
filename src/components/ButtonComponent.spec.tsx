/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import ButtonComponent from './ButtonComponent';

describe('Contributors', () => {
  it('should render the Welcome Screen', () => {
    const tree = render(<ButtonComponent text="test" handleClick={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
