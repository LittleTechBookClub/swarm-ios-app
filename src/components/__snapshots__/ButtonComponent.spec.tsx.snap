// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render the Button Component 1`] = `
<Text>
  Loading...
</Text>
`;
